src/bbs2gh/Services/BbsArchiveDownloader.cs

```diff
@@ -0,0 +1,97 @@
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using Renci.SshNet;
+
+namespace OctoshiftCLI.BbsToGithub.Services;
+
+public class BbsArchiveDownloader : IDisposable
+{
+    private const int DOWNLOAD_PROGRESS_REPORT_INTERVAL_IN_SECONDS = 10;
+    private const string DEFAULT_BBS_SHARED_HOME_DIRECTORY = "/var/atlassian/application-data/bitbucket/shared";
+    private const string EXPORT_ARCHIVE_SOURCE_DIRECTORY = "data/migration/export";
+
+    private readonly ISftpClient _sftpClient;
+    private readonly OctoLogger _log;
+    private DateTime _nextProgressReport;
+
+    internal FileSystemProvider FileSystemProvider = new();
```

Comment:
why don't we inject this?

Reply:
I wanted to keep it simple because it is only supposed to be used for tests and it's not even visible to other assemblies (except our unit tests one). But if you think it's better to inject it I'll do so.

Reply:
addressed in 508e086bbd64c5766ba77cdd00b690d1ed1ec286